### PR TITLE
 MULE-14912: Extension model should expose name and config-ref parameters

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/NoConfigRefFoundException.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.manager;
+
+import org.mule.runtime.api.meta.model.ComponentModel;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+
+import static java.lang.String.format;
+
+public class NoConfigRefFoundException extends IllegalStateException {
```
Comment (estebanwasinger) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/NoConfigRefFoundException.java]: jdoc

File: tests/runner/src/main/java/org/mule/test/runner/api/ExtensionPluginMetadataGenerator.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ public ErrorTypeRepository getErrorTypeRepository() {
     });
     try {
       extensionManager.initialise();
-    } catch (InitialisationException e) {
+    } catch (Exception e) {
```
Comment (estebanwasinger) [inline tests/runner/src/main/java/org/mule/test/runner/api/ExtensionPluginMetadataGenerator.java]: why is required to change it?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessorBuilder.java [hunk 3]
```diff
@@ -147,4 +147,10 @@ protected ResolverSet getArgumentsResolverSet() throws ConfigurationException {
     return this;
   }
 
+  protected ConfigurationProvider getConfigurationProvider() {
```
Comment (estebanwasinger) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessorBuilder.java]: Could the ConfigurationProvider change between calls?
If not, the result can be cached.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManager.java [hunk 4]
```diff
@@ -81,7 +82,7 @@
   private ExtensionActivator extensionActivator;
 
   @Override
-  public void initialise() throws InitialisationException {
+  public void initialise() {
```
Comment (estebanwasinger) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManager.java]: You are throwing exceptions, make them throw `InitialisationException` if not, you are not compliant with the API.

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java [hunk 5]
```diff
@@ -164,53 +118,49 @@
...
+   * @param builder     a {@link Builder} used as a prototype to generate new definitions
+   * @param dslResolver a {@link DslSyntaxResolver} instance associated with the {@link ExtensionModel} being parsed
+   * @param ctx         the {@link ExtensionParsingContext} in which {@code this} parser operates
    */
-  protected ExtensionDefinitionParser(Builder baseDefinitionBuilder, DslSyntaxResolver dslSyntaxResolver,
-                                      ExtensionParsingContext parsingContext) {
-    this.baseDefinitionBuilder = baseDefinitionBuilder;
-    this.dslResolver = dslSyntaxResolver;
-    this.parsingContext = parsingContext;
+  protected ExtensionDefinitionParser(Builder builder, DslSyntaxResolver dslResolver, ExtensionParsingContext ctx) {
```
[the remark this answers is not in the document]
Comment (juandesi) [reply modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java]: prettynes

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/config/ConfigurationProviderObjectFactory.java [hunk 6]
```diff
@@ -86,7 +84,7 @@ private ConfigurationProvider createInnerInstance() throws ConfigurationExceptio
       try {
         if (resolverSet.isDynamic() || connectionProviderResolver.isDynamic()) {
           configurationProvider =
-              configurationProviderFactory.createDynamicConfigurationProvider(name, extensionModel,
+              configurationProviderFactory.createDynamicConfigurationProvider(getName(), extensionModel,
```
[the remark this answers is not in the document]
Comment (juandesi) [reply modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/config/ConfigurationProviderObjectFactory.java]: how can this be cached?, this need to be resolved for each instance every time

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (estebanwasinger) [reply modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java]: just changed argument names and not field names, I prefer consistency

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/config/ConfigurationProviderObjectFactory.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (estebanwasinger) [reply modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/config/ConfigurationProviderObjectFactory.java]: use a lazy value as an field instance

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/config/ConfigurationProviderObjectFactory.java [hunk 7]
```diff
@@ -14,24 +14,25 @@
...
 import org.mule.runtime.module.extension.internal.config.dsl.AbstractExtensionObjectFactory;
 import org.mule.runtime.module.extension.internal.runtime.config.ConfigurationProviderFactory;
 import org.mule.runtime.module.extension.internal.runtime.config.DefaultConfigurationProviderFactory;
-import org.mule.runtime.module.extension.internal.runtime.resolver.ConnectionProviderResolver;
-import org.mule.runtime.module.extension.internal.runtime.resolver.ConnectionProviderValueResolver;
-import org.mule.runtime.module.extension.internal.runtime.resolver.ImplicitConnectionProviderValueResolver;
-import org.mule.runtime.module.extension.internal.runtime.resolver.ResolverSet;
-import org.mule.runtime.module.extension.internal.runtime.resolver.StaticConnectionProviderResolver;
+import org.mule.runtime.module.extension.internal.runtime.exception.RequiredParameterNotSetException;
+import org.mule.runtime.module.extension.internal.runtime.resolver.*;
```
[the remark this answers is not in the document]
Comment (estebanwasinger) [reply modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/config/ConfigurationProviderObjectFactory.java]: ☝️

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/resolver/ValueResolverFactory.java [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    return resolver;
+  }
+
+  private boolean isRequiredByExclusiveOptional(Set<ModelProperty> modelProperties) {
+    return modelProperties.stream().anyMatch(modelProperty -> modelProperty instanceof ExclusiveOptionalModelProperty
+        && ((ExclusiveOptionalModelProperty) modelProperty).isOneRequired());
+  }
+
```
Comment (estebanwasinger) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/resolver/ValueResolverFactory.java]: A lot of empty lines

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/resolver/ValueResolverFactory.java [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+  private ValueResolver getValueResolverFromMetadataType(String paramName, MetadataType expected, Object value,
+                                                         Object defaultValue, boolean acceptsReferences, Class<?> expectedClass) {
+    ValueResolverFactoryTypeVisitor visitor =
+        new ValueResolverFactoryTypeVisitor(dslResolver, paramName, expected, value, defaultValue, acceptsReferences,
+                                            expectedClass);
+    expected.accept(visitor);
+    return visitor.getResolver();
+  }
+
```
Comment (estebanwasinger) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/resolver/ValueResolverFactory.java]: Same

File: modules/extensions-xml-support/src/test/resources/modules/model/module-stereotypes.json [hunk 10]
```diff
@@ -348,6 +369,55 @@
...
                   "description": "",
                   "modelProperties": {}
+                },
+                {
+                  "type": {
+                    "format": "java",
+                    "type": "Object",
+                    "annotations": {
+                      "typeId": "org.mule.runtime.extension.api.runtime.config.ConfigurationProvider",
+                      "classInformation": {
```
Comment (estebanwasinger) [inline modules/extensions-xml-support/src/test/resources/modules/model/module-stereotypes.json]: This wasn't to be de deleted?

Comment (juandesi) [reply modules/extensions-xml-support/src/test/resources/modules/model/module-stereotypes.json]: yes, I dont know why still there 🤔

**Comment by Bardammu** [comment]
--validate

**Pull request merged**